res/controllers/midi-components-0.0.js

```diff
@@ -301,8 +309,48 @@
         shift: function () {
             this.inKey = 'hotcue_' + this.number + '_clear';
         },
+        getColor: function() {
+            if (this.colorIdKey !== undefined) {
+                return color.predefinedColorFromId(engine.getValue(this.group,this.colorIdKey));
+            } else {
+                return null;
+            }
+        },
+        output: function(value) {
+            var outval = this.outValueScale(value);
+            if (this.colorIdKey !== undefined && outval !== this.off) {
```

Comment:
Is `outval !== this.off` necessary here? Shouldn't `outputColor` be able to handle sending a MIDI signal to turn the LED off?

Reply:
`outputColor` takes a color ID as argument, and there is no color id to represent the state "off". There is the "no color", but that is for hot cues that are enabled but don't have color assigned to them. Turning off LEDs could be moved into the outputColor function and trigged if the ID parameter is `undefined` or something like that, but that would be messy IMHO.

Reply:
That's exactly what I was thinking as well.

Reply:
Okay, thanks for explaining. Adding a comment above the `if` line saying something like "There is no color ID for the off state, so do not use outputColor for that."

Reply:
I can't think of a single controller that sets RGB leds discrete midi messages for every channel. I only know controllers that either just have a palette that can be chosen from via a single midi message or just set the RGB values in one go via a SysEx message.
In the end. I don't know a single controller whose led can't be turned of via a `note-off`` or 0 value. If you can point me to one whose leds can't be controlled otherwise, I will worry about that case, but otherwise, I don't think your case exists...

Reply:
I think you are correct. Add the comment, then I think this is ready to merge.

Reply:
Sorry for the late response. I can't really think of a comment to put there that is short but still explains the Issue. Do you have something in mind?

Reply:
How about "There is no color ID for the off state, so do not use the outputColor function for that."

Reply:
ok. will do.